project.py

```
@@ -23,6 +46,33 @@ def process_pdf_to_record(path: str) -> Dict[str, object]:
+def grobid_generate_tei(path: str, tei_dir: str = "data/xml") -> List[str]:
+    """
+    Generate TEI XML files for a single PDF or a directory of PDFs.
+    Returns list of saved TEI file paths.
+    """
+    from paperslicer.grobid.ingest import GrobidIngestor
+
+    ing = GrobidIngestor(tei_dir=tei_dir)
+    if not ing.client.is_available():
+        raise SystemExit(
+            "GROBID not available. Start it and set GROBID_URL, or use Docker."
+        )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)